plaso/parsers/bsm.py

```diff
@@ -71,1118 +36,447 @@ def __init__(self):
+      if token_data.data_format == 4:
+        data = bytes(bytearray(token_data.data)).split(b'\x00')[0]
+        data = data.decode('utf-8')
+      else:
+        data = ''.join(['{0:02x}'.format(byte) for byte in token_data.data])
+      return {
+          'format': format_string,
+          'data': data}
+
+    elif token_type == 0x22:
+      return {
+          'object_type': token_data.object_type,
+          'object_id': token_data.object_identifier}
+
+    elif token_type == 0x23:
+      return {'path': token_data.path.rstrip('\x00')}
+
+    elif token_type in (0x24, 0x26, 0x75, 0x77, 0x7a, 0x7b, 0x7c, 0x7d):
+      if token_type in (0x24, 0x26, 0x75, 0x77) or token_data.net_type == 4:
+        ip_address = self._FormatPackedIPv4Address(token_data.ip_address)
+      elif token_data.net_type == 16:
+        ip_address = self._FormatPackedIPv6Address(token_data.ip_address)
+      else:
+        ip_address = 'unknown'
+
+      return {
+          'aid': token_data.audit_user_identifier,
+          'euid': token_data.effective_user_identifier,
+          'egid': token_data.effective_group_identifier,
+          'uid': token_data.real_user_identifier,
+          'gid': token_data.real_group_identifier,
+          'pid': token_data.process_identifier,
+          'session_id': token_data.session_identifier,
+          'terminal_port': token_data.terminal_port,
+          'terminal_ip': ip_address}
+
+    elif token_type in (0x27, 0x52, 0x72):
+      error_string = bsmtoken.BSM_ERRORS.get(token_data.status, 'UNKNOWN')
+      return {
+          'error': error_string,
+          'token_status': token_data.status,
+          'call_status': token_data.return_value}
+
+    elif token_type == 0x28:
+      return {'text': token_data.text.rstrip('\x00')}
+
+    elif token_type == 0x29:
+      data = ''.join(['{0:02x}'.format(byte) for byte in token_data.data])
+      return {'data': data}
+
+    elif token_type == 0x2a:
+      ip_address = self._FormatPackedIPv4Address(token_data.ip_address)
+      return {'ip': ip_address}
+
+    elif token_type == 0x2b:
+      data = ''.join(['{0:02x}'.format(byte) for byte in token_data.data])
+      return {'IPv4_Header': data}
+
+    elif token_type == 0x2c:
+      return {'port_number': token_data.port_number}
+
+    elif token_type in (0x2d, 0x71):
+      return {
+          'string': token_data.argument_value.rstrip('\x00'),
+          'num_arg': token_data.argument_index,
+          'is': token_data.argument_name}
+
+    elif token_type == 0x2f:
+      return {'sequence_number': token_data.sequence_number}
+
+    elif token_type == 0x32:
+      return {
+          'user_id': token_data.user_identifier,
+          'group_id': token_data.group_identifier,
+          'creator_user_id': token_data.creator_user_identifier,
+          'creator_group_id': token_data.creator_group_identifier,
+          'access': token_data.access_mode}
+
+    # elif token_type in (0x34, 0x3b):
```

Comment:
Why is this commented out? Does this need a todo, or to be fixed?

Reply:
This is still WIP.